frontend/src/app/components/enterprise/enterprise-trial.service.ts

```diff
@@ -0,0 +1,155 @@
+import {Injectable} from "@angular/core";
+import {I18nService} from "core-app/modules/common/i18n/i18n.service";
+import {HttpClient, HttpErrorResponse} from "@angular/common/http";
+import {PathHelperService} from "core-app/modules/common/path-helper/path-helper.service";
+import {NotificationsService} from "core-app/modules/common/notifications/notifications.service";
+import {FormBuilder, FormControl, FormGroup} from "@angular/forms";
+
+
+@Injectable()
+export class EnterpriseTrialService {
+  private localStorageKey = 'openProject-enterprise-trial';
+
+  public text = {
+    resend_success: this.I18n.t('js.admin.enterprise.trial.resend_success'),
+    resend_warning: this.I18n.t('js.admin.enterprise.trial.resend_warning')
+  };
+
+  public savedUserData:any; // saved data in local storage
+  public trialForm:FormGroup;
+  public baseUrlAugur = 'https://augur.openproject-edge.com';
+  public trialLink:string;
+  public resendLink:string;
+
+  public confirmed:boolean;
+  public cancelled = false;
+  public status:string|undefined;
```

Comment:
Please specify the types the `status` variable can be. So `public status:'startTrial'|'mailSubmitted'|undefined`.